tests/trainer/test_trainer.py

```diff
@@ -77,38 +77,40 @@ def test_trainer_validation(mosaic_trainer_hparams: TrainerHparams, invalid_hpar
 
 
 @pytest.mark.filterwarnings("ignore:Deterministic mode is activated:UserWarning")
-def test_trainer_determinism(mosaic_trainer_hparams: TrainerHparams, ddp_tmpdir: str):
+@pytest.mark.timeout(90)
+def test_trainer_determinism(mosaic_trainer_hparams: TrainerHparams):
     mosaic_trainer_hparams.seed = 10
     mosaic_trainer_hparams.deterministic_mode = True
     mosaic_trainer_hparams.max_epochs = 2
 
-    mosaic_trainer_hparams.ddp.store = FileStoreHparams(os.path.join(ddp_tmpdir, "first_store"))
     first_trainer = Trainer.create_from_hparams(mosaic_trainer_hparams)
     first_trainer.fit()
     first_model = first_trainer.state.model.module
     assert isinstance(first_model, BaseMosaicModel)
     assert first_trainer.state.train_dataloader is not None
-    first_loss = get_total_loss(first_model, first_trainer.state.train_dataloader)
+    first_loss = get_total_loss(first_model, first_trainer.state.train_dataloader, first_trainer.ddp)
+
+    # Need to reinitialize some distributed settings in order to train twice in the same process
+    torch.distributed.destroy_process_group()
```

Comment:
Curious why wasn't the process group destroyed properly the first time?

Reply:
There's just nowhere for it to be destroyed. I don't think the trainer itself handles destroying its own process group.